ui/lib/core/addon/components/page/error.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { ResponseError } from '@hashicorp/vault-client-typescript';
+
+import type Owner from '@ember/owner';
+import type ApiService from 'vault/services/api';
+
+interface Args {
+  error: ResponseError | Error;
+}
+
+type Error = {
+  httpStatus?: number;
+  path?: string;
+  message: string;
+  errors: string[];
+};
+
+export default class PageErrorComponent extends Component<Args> {
+  @service declare readonly api: ApiService;
+
+  @tracked declare error: Error;
+
+  constructor(owner: Owner, args: Args) {
+    super(owner, args);
+    this.unpackError();
+  }
+
+  async unpackError() {
+    if (this.args.error instanceof ResponseError) {
+      const { status, path, response } = await this.api.parseError(this.args.error);
```

Comment:
Any chance we could also error on this.api.parseError? If so we should wrap in try/catch and indicate we're swallowing the error.

Reply:
The async bit of this is calling `json` on the response so it shouldn't unless the json is invalid. In that case I would put the try/catch in the `parseError` method and return a generic error. I will test this out with mirage to see if it's possible to make it error and follow up in another PR if necessary.